android/src/main/java/io/radar/react/RNRadarModule.java

```diff
@@ -331,41 +331,158 @@ public void rejectEvent(String eventId) {
     }
 
     @ReactMethod
-    public void startTrip(ReadableMap optionsMap) {
+    public void startTrip(ReadableMap optionsMap, final Promise promise) {
         try {
             JSONObject optionsObj = RNRadarUtils.jsonForMap(optionsMap);
             RadarTripOptions options = RadarTripOptions.fromJson(optionsObj);
             Radar.startTrip(options, new Radar.RadarTripCallback() {
                 @Override
                 public void onComplete(@NonNull Radar.RadarStatus status, @Nullable RadarTrip trip, @Nullable RadarEvent[] events) {
-
+                    if (promise == null) {
+                        return;
+                    }
+                    try {
+                        if (status == Radar.RadarStatus.SUCCESS) {
+                            WritableMap map = Arguments.createMap();
+                            map.putString("status", status.toString());
+                            if (trip != null) {
+                                map.putMap("trip", RNRadarUtils.mapForJson(trip.toJson()));
+                            }
+                            if (events != null) {
+                                map.putArray("events", RNRadarUtils.arrayForJson(RadarEvent.toJson(events)));
+                            }
+                            promise.resolve(map);
+                        } else {
+                            promise.reject(status.toString(), status.toString());
+                        }
+                    } catch (JSONException e) {
+                        Log.e(TAG, "JSONException", e);
+                        promise.reject(Radar.RadarStatus.ERROR_SERVER.toString(), Radar.RadarStatus.ERROR_SERVER.toString());
+                    }
                 }
             });
         } catch (JSONException e) {
             Log.e(TAG, "JSONException", e);
+            promise.reject(Radar.RadarStatus.ERROR_SERVER.toString(), Radar.RadarStatus.ERROR_SERVER.toString());
```

Comment:
Let's do `Radar.RadarStatus.ERROR_BAD_REQUEST.toString()` for these outer `catch`'es, where the request data is the thing throwing